src/compile/legend/encode.ts

```diff
@@ -75,9 +74,7 @@ export function symbols(fieldDef: FieldDef, symbolsSpec: any, model: UnitModel,
       {value: cfg.mark.color};
   }
 
-  if (legend.symbolColor !== undefined) {
-    symbols.fill = {value: legend.symbolColor};
-  } else if (symbols.fill === undefined) {
+  if (symbols.fill === undefined) {
```

Comment:
I tested with and without `symbols.update.fill` been specified, and it does make a difference.